src/measure_theory/set_integral.lean

```diff
@@ -0,0 +1,285 @@
+  (hgm : measurable_on s g) (hgi : integrable_on s g) (h : ∀ a, a ∈ s → f a ≤ g a) :
+  integral_on s f ≤ integral_on s g :=
+integral_on_le_integral_on_ae _ hfm hfi hgm hgi $ by filter_upwards [] h
+
+lemma integral_on_union (hsm : measurable_on s f) (hsi : integrable_on s f)
+  (htm : measurable_on t f) (hti : integrable_on t f) (h : disjoint s t) :
+  integral_on (s ∪ t) f = integral_on s f + integral_on t f :=
+by { simp only [integral_on], rw [indicator_union_of_disjoint h, integral_add hsm hsi htm hti] }
+
+lemma integral_on_union_ae (hs : is_measurable s) (ht : is_measurable t) (hsm : measurable_on s f)
+  (hsi : integrable_on s f) (htm : measurable_on t f) (hti : integrable_on t f) (h : ∀ₘ a, a ∉ s ∩ t) :
+  integral_on (s ∪ t) f = integral_on s f + integral_on t f :=
+begin
+  simp only [integral_on],
+  have := integral_congr_ae _ _ (indicator_union_ae h f),
+  rw [this, integral_add hsm hsi htm hti],
+  { exact hsm.union hs ht htm },
+  { exact hsm.add htm }
+end
+
+lemma tendsto_integral_on_of_monotone {s : ℕ → set α} {f : α → β} (hsm : ∀i, is_measurable (s i))
+  (h_mono : monotone s) (hfm : measurable_on (Union s) f) (hfi : integrable_on (Union s) f) :
+  tendsto (λi, integral_on (s i) f) at_top (nhds (integral_on (Union s) f)) :=
+let bound : α → ℝ := indicator (Union s) (λa, ∥f a∥) in
+begin
+  apply tendsto_integral_of_dominated_convergence,
+  { assume i, exact hfm.subset (hsm i) (subset_Union _ _) },
+  { assumption },
+  { show integrable_on (Union s) (λa, ∥f a∥), rwa integrable_on_norm_iff },
+  { assume i, apply all_ae_of_all,
+    assume a,
+    rw [norm_indicator_eq_indicator_norm],
+    exact indicator_le_indicator_of_subset (subset_Union _ _) (λa, norm_nonneg _) _ },
+  { filter_upwards [] tendsto_indicator_of_monotone _ h_mono _ }
+end
+
+lemma tendsto_integral_on_of_decreasing (s : ℕ → set α) (f : α → β) (hsm : ∀i, is_measurable (s i))
+  (h_mono : ∀i j, i ≤ j → s j ⊆ s i) (hfm : measurable_on (s 0) f) (hfi : integrable_on (s 0) f) :
+  tendsto (λi, integral_on (s i) f) at_top (nhds (integral_on (Inter s) f)) :=
+let bound : α → ℝ := indicator (s 0) (λa, ∥f a∥) in
+begin
+  apply tendsto_integral_of_dominated_convergence,
+  { assume i, refine hfm.subset (hsm i) (h_mono _ _ (zero_le _)) },
+  { exact hfm.subset (is_measurable.Inter hsm) (Inter_subset _ _) },
+  { show integrable_on (s 0) (λa, ∥f a∥), rwa integrable_on_norm_iff },
+  { assume i, apply all_ae_of_all,
+    assume a,
+    rw [norm_indicator_eq_indicator_norm],
+    refine indicator_le_indicator_of_subset (h_mono _ _ (zero_le _)) (λa, norm_nonneg _) _ },
+  { filter_upwards [] tendsto_indicator_of_decreasing _ h_mono _ }
+end
+
+-- TODO : prove the following proposition
+
+-- lemma integral_on_Union (s : ℕ → set α) (f : α → β) (hm : ∀i, is_measurable (s i))
+--   (hd : pairwise (disjoint on s)) (hf : integrable_on (Union s) f) :
+--   integral_on (Union s) f = ∑i, integral_on (s i) f := sorry
```

Comment:
One more TODO (may be for later): `integral_on s f` is the usual integral of `f` w.r.t. the restriction of the measure to `s`.

Reply:
I still don't know how to prove this. I think I may need to generalize the dominated convergence theorem to types other than `nat`.

Reply:
I think that we should merge your PR with this TODO. I have a generalized dominated convergence theorem (with `[encodable ι] {l : filter ι} (hl : l ≠ ⊥)` assumptions) almost ready but if we merge it here, then someone will have to review the union.

Reply:
Oh, that's very good!

Reply:
It seems that the difference between "almost ready" and "ready" is larger than I expected. I'll add more details tonight or tomorrow.

Reply:
The restriction to countable `ι` seems a bit strong.  AFAIK you can do the dominated convergence theorem for any filter with a countable basis, e.g. neighborhood filters in a first-countable space.  I looked into this a while back when we didn't have the Bochner integral yet: https://github.com/gebner/mathlib/commit/4cb637c6c1d2f1ebc2744c9fd436529062ed198b#diff-2d55094f133203e31652b047c7f4e6e2R965-R972

Reply:
@gebner Thank you! Are you going to PR this?

Reply:
I won't have time for it this week, but I could clean it up next week.  Feel free to copy the 20 lines if you want it sooner.

Reply:
I've got everything cleaned up and proved the same thing for the Bochner integral. I think I'll PR it if you don't mind.

Reply:
:+1: